[WebGPU] Move getPreferredFormat() from GPUCanvasContext to GPU to align with spec

tuankiet65 opened the pull request:
#### 1e3ea2a7462c8e91954dd17a7e25129c992c86bd
<pre>
[WebGPU] Move getPreferredFormat() from GPUCanvasContext to GPU to align with spec
<a href="https://bugs.webkit.org/show_bug.cgi?id=242648">https://bugs.webkit.org/show_bug.cgi?id=242648</a>

Reviewed by Myles C. Maxfield.

* Source/WebCore/Modules/WebGPU/GPU.cpp:
(WebCore::GPU::getPreferredCanvasFormat):
* Source/WebCore/Modules/WebGPU/GPU.h:
* Source/WebCore/Modules/WebGPU/GPU.idl:
* Source/WebCore/Modules/WebGPU/GPUCanvasContext.cpp:
(WebCore::GPUCanvasContext::getPreferredFormat): Deleted.
* Source/WebCore/Modules/WebGPU/GPUCanvasContext.h:
* Source/WebCore/Modules/WebGPU/GPUCanvasContext.idl:

Canonical link: <a href="https://commits.webkit.org/252658@main">https://commits.webkit.org/252658@main</a>
</pre>

djg reviewed (approved):
LGTM

litherum commented on Source/WebCore/Modules/WebGPU/GPU.cpp:
```diff
@@ -63,4 +63,9 @@ void GPU::requestAdapter(const std::optional<GPURequestAdapterOptions>& options,
 
+GPUTextureFormat GPU::getPreferredCanvasFormat()
+{
+    return GPUTextureFormat::Rgba8unorm;
```
This probably should be plumbed through `PAL`, no?

Also, I believe our preferred format is BGRA8.

tuankiet65 replied:
Yeah, the long term solution is to plumb it through PAL. And I have it as BGRA8 on my local build, I'm not changing it here because I like commits that do one thing only.

tuankiet65 committed:
[WebGPU] Move getPreferredFormat() from GPUCanvasContext to GPU to align with spec
https://bugs.webkit.org/show_bug.cgi?id=242648

Reviewed by Myles C. Maxfield.

* Source/WebCore/Modules/WebGPU/GPU.cpp:
(WebCore::GPU::getPreferredCanvasFormat):
* Source/WebCore/Modules/WebGPU/GPU.h:
* Source/WebCore/Modules/WebGPU/GPU.idl:
* Source/WebCore/Modules/WebGPU/GPUCanvasContext.cpp:
(WebCore::GPUCanvasContext::getPreferredFormat): Deleted.
* Source/WebCore/Modules/WebGPU/GPUCanvasContext.h:
* Source/WebCore/Modules/WebGPU/GPUCanvasContext.idl:

Canonical link: https://commits.webkit.org/252658@main
